### PR TITLE
DWARF: high_pc computation

### PR DESCRIPTION
**Pull request by kripken**
Update `high_pc` values. These are interesting as they
may be a relative offset compared to the `low_pc`.

For functions we already had both a start and an end. Add
such tracking for instructions as well.

This PR depends on #2594 (testcase updates).

### CODE REVIEW

File: src/wasm/wasm-debug.cpp [hunk 1]
```diff
@@ -427,10 +441,21 @@ struct LocationUpdater {
   // address, or if there was but if that instruction no longer exists, return
   // 0. Otherwise, return the new updated location.
   uint32_t getNewExprAddr(uint32_t oldAddr) const {
-    if (auto* expr = oldExprAddrMap.get(oldAddr)) {
+    if (auto* expr = oldExprAddrMap.getStart(oldAddr)) {
```
Comment (dschuff) [inline src/wasm/wasm-debug.cpp]: update the comment on line 432-434

File: test/passes/fannkuch3.bin.txt [hunk 2]
```diff
@@ -2534,7 +2534,7 @@ Abbrev table for offset: 0x00000000
 
 0x00000082:   DW_TAG_subprogram [10] *
                 DW_AT_low_pc [DW_FORM_addr]	(0x0000000000000006)
-                DW_AT_high_pc [DW_FORM_data4]	(0x00000397)
+                DW_AT_high_pc [DW_FORM_data4]	(0x00000383)
```
Comment (dschuff) [inline test/passes/fannkuch3.bin.txt]: I guess this means that Binaryen shrank the function? Is that expected for this test?

**Comment by kripken** [comment]
We don't seem to have the absolute addr variant - perhaps llvm never emits it? (It does seem less efficient, so seems reasonable it wouldn't.

File: test/passes/fannkuch3.bin.txt (hunk 2 above)
Comment (kripken) [reply test/passes/fannkuch3.bin.txt]: Yes, it shrinks it a little even without running optimizations, due to LEBs being compressed (which clang doesn't do in all places). I verified that size is correct.

File: src/wasm/wasm-debug.cpp (hunk 1 above)
Comment (kripken) [reply src/wasm/wasm-debug.cpp]: comment updated.

**Pull request merged**